src/main/com/example011/radek/v1/Mines.java

```diff
@@ -0,0 +1,138 @@
+package main.com.example011.radek.v1;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Scanner;
+
+public class Mines {
+
+    protected int playgroundSize;
+
+    protected int numberOfMines;
+
+    protected String[][] playground;
+
+    public Mines(int size, int mines) {
+
+        if (checkPlaygroundSize(size)) {
+            setPlaygroundSize(size);
+            setNumberOfMines(mines);
+        } else {
+            System.out.println("Bad size of playground, size must be bigger then 5!");
+            System.exit(0);
+        }
+    }
+
+    public void fillPlaygroundWithMines() throws NoSuchAlgorithmException {
+
+        playground = new String[playgroundSize][playgroundSize];
+        SecureRandom random = SecureRandom.getInstanceStrong();
+        int placedMines = 0;
+        while (placedMines != getNumberOfMines() || placedMines == 0) {
+            int generatedRowNumber = random.nextInt(playgroundSize);
+            int generatedColumnNumber = random.nextInt(playgroundSize);
+            if (playground[generatedRowNumber][generatedColumnNumber] == null) {
+                playground[generatedRowNumber][generatedColumnNumber] = StringConstant.MINE.getConstant();
+                placedMines++;
+            }
+        }
+    }
+
+    public void printPlayground(boolean showMines) {
+
+        System.out.println("Generating playground...");
+        StringBuilder playingField = new StringBuilder(playgroundSize * playgroundSize * 3);
+        for (int i = 0; i < playgroundSize; i++) {
+            for (int j = 0; j < playgroundSize; j++) {
+                if (playground[i][j] != null && ((showMines && playground[i][j].equals("x")) || playground[i][j].equals("o"))) {
+                    playingField.append(playground[i][j])
+                            .append(StringConstant.TABULATOR.getConstant());
+                } else {
+                    playingField.append(StringConstant.EMPTY.getConstant())
+                            .append(StringConstant.TABULATOR.getConstant());
+                }
+            }
+            playingField.append(StringConstant.NEW_LINE.getConstant());
+        }
+        System.out.println(playingField);
+    }
+
+    public void getSizeAndMines() {
+
+        System.out.println("Playground size is: " + playgroundSize + " and on this ground you can find: " + numberOfMines + " mines.");
+    }
+
+    private void setPlaygroundSize(int size) {
+
+        playgroundSize = size;
+    }
+
+    private boolean checkPlaygroundSize(int size) {
```

Comment:
1) nazev metody rika zkontrolujHraciPlochu ... lepsi nazev ... isPlaygroundSizeBigger or Smaller nez neco ... vubec nevadi, ze to bude dlouhe, ale at je to co nejvice konkretnejsi, pouvazuj jak
2) ta konctanta 5 urcite ne ... mas to jako vnitrni atribut, tak ho řádně využij .. necheme zadne magic numbers

Reply:
Hotovo

Reply:
jeste jedna vec, pokud to je konstanta, coz v nasem pripade je, tak to muze byt rovnou static, tzn. je to pevne vytvoreno a nebude se to nijak menit, ... tim ze to je static, tak to pak muze byt i pojmenovano UPPERCASE spravne, jak jsi pouzil pro konstantu 

v konstruktoru pouzij ve vypisu hodnotu z konstanty, nahodou zmenis možné minimum a v textu na to pak zapomnees(aktualne tam mas napevno string)

Reply:
Hotovo, udělal jsem to přes StringBuilder, což by mělo být čistější, jestli tomu rozumím správně

Reply:
slo hlavne o to tam pouzit tu constantu, coz jsi pochopil a stringbuilder je tady troskku overkill ale na druhou stranu zase ne, ale toto je ok, jde o to procvicit ... dalsi moznosti bylo pouzit Stirng.concat nebo to normalne nacpat v jednom outu .
System.our.ptint("xxx " + MINIMUM + "!") taky citelne